src/main/java/com/google/songgame/data/YoutubeParser.java

```diff
@@ -0,0 +1,161 @@
+import com.google.api.client.json.JsonFactory;
+import com.google.api.client.json.jackson2.JacksonFactory;
+import com.google.api.services.youtube.YouTube;
+import com.google.api.services.youtube.YouTubeRequestInitializer;
+import com.google.api.services.youtube.model.VideoListResponse;
+import com.google.api.services.youtube.model.Video;
+import com.google.api.services.youtube.model.PlaylistItemListResponse;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Text;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Collection;
+import java.util.Random;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.api.client.json.gson.GsonFactory;
+
+public final class YoutubeParser {
+
+  private static final String DEVELOPER_KEY = "AIzaSyBZw4Z25Lect7ux9z960RCM7YORcYo6slc";
+  private static final String APPLICATION_NAME = "Song Guessing Game";
+  private static final JsonFactory GSON_FACTORY = GsonFactory.getDefaultInstance();
+  private static final long MAX_RESULTS = 25L;
+
+  public ArrayList<String> getPlaylistVideoIds(String playlistUrl) {
+    String playlistId = getPlaylistIdFromUrl(playlistUrl);
+    PlaylistItemListResponse playlistItem = new PlaylistItemListResponse();
+    // Retrieve Playlist item from Youtube API
+    try {
+      playlistItem = getPlaylistInfo(playlistId);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+    // Parse Playlist item Json string to retrieve video IDs
+    String playlistItemJson = new Gson().toJson(playlistItem);
+    ArrayList<String> playlistVideoIds = parseVideoIdsFromPlaylistItem(playlistItemJson);
+    return playlistVideoIds;
+  }
+
+  /**
+   * Returns playlist ID
+   */
+  private String getPlaylistIdFromUrl(String playlistUrl) {
+    if (playlistUrl.contains("youtube.com/playlist?list=")) {
+      int start = playlistUrl.indexOf("list=") + 5;
+      int end = playlistUrl.length();
+      if (playlistUrl.contains("&")) {
+        end = playlistUrl.indexOf("&", start);
+      }
+      return playlistUrl.substring(start, end);
+    } else {
+      System.err.println(playlistUrl + " is not a valid YouTube Playlist URL.");
+      return "";
+      // TODO @hdee: do something more complicated to handle this error.
+      // TODO @hdee: test this method.
+    }
+  }
+
+   /**
```

Comment:
I think the formatting is slightly off here. Would you mind running the formatter on this file?